saw-core-what4/src/SAWCoreWhat4/What4.hs

```diff
@@ -970,14 +971,16 @@ Value for `f x`:
   ([ w8s ! 0, w8s ! 1, w8s ! 2, w8s ! 3 ], bools ! 0)
 -}
 
--- | This is the state we use to generate uninterpreted values.
+
+
+-- | Track how many uninterpreted results we need for each base type.
+newtype UnintCount (tc :: BaseType) = UnintCount Natural
+
+-- | Uninterpreted function results for each base type.
 type UnintState sym = MapF BaseTypeRepr (Arr sym)
 
--- | For each base type we track the name of the array to project from and
--- the next index.
-data Arr sym tc = Arr
-  (SymExpr sym (BaseArrayType (Ctx.EmptyCtx Ctx.::> BaseIntegerType) tc))
-  !Integer
+-- | Uninterpreted function results for a particular base type.
+data Arr sym tc = Arr [SymExpr sym tc]
```

Comment:
It might be helpful to state the relationship between the `[SymExpr sym tc]` list's length and the `UnintCount` for that `BaseType`.